hw/ip/edn/data/edn.hjson

```diff
@@ -0,0 +1,264 @@
+          name: "CMD_ACK",
+          desc: '''This bit is set when the command request has completed. This bit must
+                   be cleared before a new request is made.
+                '''
+        }
+        { bits: "4",
+          name: "CMD_STS",
+          desc: ''' 
+                This one bit field is the status code returned with the application command ack.
+                0b0: Request completed successfully
+                0b1: Request completed with an error
+                '''
+        }
+      ]
+    },
+    {
+      name: "RESEED_CMD",
+      desc: "EDN csrng reseed command register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwext: "true",
+      hwqe: "true",
+      regwen: "REGEN",
+      fields: [
+        { bits: "31:0",
+          name: "RESEED_CMD",
+          desc: '''
+                Writing this register will fill a FIFO with up to 13 command words (32b words).
+                This FIFO will be used to automatically send out a reseed command to the CSRNG 
+                appl interface when in AUTO_REQ_MODE. This command will be sent only after 
+                the MAX_NUM_REQS_BETWEEN_RESEEDS counter value has reached zero.
+                '''
+        }
+      ]
+    },
+    {
+      name: "GENERATE_CMD",
+      desc: "EDN csrng generate command register",
+      swaccess: "wo",
+      hwaccess: "hro",
+      hwext: "true",
+      hwqe: "true",
+      regwen: "REGEN",
+      fields: [
+        { bits: "31:0",
+          name: "GENERATE_CMD",
+          desc: '''
+                Writing this register will fill a FIFO with up to 13 command words (32b words).
+                This FIFO will be used to automatically send out a generate command to the CSRNG 
+                appl interface when in AUTO_REQ_MODE. This command will be sent only after 
+                receiving a command ack from the previous command.
+                '''
+        }
+      ]
+    },
+    { name: "MAX_NUM_REQS_BETWEEN_RESEEDS",
+      desc: "EDN maximum number of requests between reseeds register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      hwqe: "true",
+      fields: [
+        {
+            bits: "31:0",
+            name: "MAX_NUM_REQS_BETWEEN_RESEEDS",
+            desc: '''
+                  Setting this field will set the number of generate requests that can be made 
+                  to CSRNG before a reseed request is made. This value only has meaning when in
+                  AUTO_REQ_MODE. A value of 2^48 is allowed, but this register only supports 
+                  a value of 2^32 requests. This register will be used by a counter that counts
+                  down, that when reaching zero will issue a reseed command to CSRNG.
+                  '''
+        },
+      ]
+    },
+    { name: "BROADCAST_REQ_TIMER",
+      desc: "EDN broadcast mode request timer register",
+      swaccess: "rw",
+      hwaccess: "hro",
+      fields: [
+        {
+            bits: "31:0",
```

Comment:
I think i'm a bit confused, I had assumed this timer would control the broadcast frequency.. and that when the broadcast entropy bits were drained, it would make a new request to csrng.  Sounds like this is not the case?

If so, how is the broadcast frequency controlled..? is it just pumped out cycle after cycle until its drained... then we wait until the next request before more are pumped out?

Reply:
Sorry that this is not clear, but there are a number of factors here. The generate command itself can have N number of words requested per command. If N=1, then the broadcast timer has more weight on control, verses if N was 100. Once a word is returned to EDN, all 128 bits are metered out on whatever the bus size is, say 32 bits. So if the requesting peripheral takes 32 and quits, the other bits remain in EDN queued up. But once the full 128 bits are consumed, then the EDN logic will fill its packer at the next opportunity.  It may be a little confusion only because of, again, the flexibility. If we decide to constraint the function some, then less so. But this all cheap and still kind of simple, so why bother.

Reply:
sorry Mark, what do you mean by "more weight on control"?

I think what you're saying is that the EDN in broadcast mode will meter out all of the bits it has available (is this on back to back cycles without gaps?) Once that is complete, it will use this timer to determine when it should next request for more?

Reply:
Hi Tim, 
   Mark can weigh in here, but my understanding is that the broadcast interface gets at most one burst of values in every timed interval.  If the packer still has values left over those are saved for later.